support-frontend/assets/components/directDebit/directDebitForm/directDebitFormContainer.tsx

```diff
@@ -0,0 +1,106 @@
+import { Recaptcha } from 'components/recaptcha/recaptcha';
+import {
+	setAccountHolderConfirmation,
+	setAccountHolderName,
+	setAccountNumber,
+	setFormError,
+	setSortCodeString,
+} from 'helpers/redux/checkout/payment/directDebit/actions';
+import {
+	confirmAccountDetails,
+	directDebitErrorMessages,
+} from 'helpers/redux/checkout/payment/directDebit/thunks';
+import {
+	expireRecaptchaToken,
+	setRecaptchaToken,
+} from 'helpers/redux/checkout/recaptcha/actions';
+import {
+	useContributionsDispatch,
+	useContributionsSelector,
+} from 'helpers/redux/storeHooks';
+import { trackComponentLoad } from 'helpers/tracking/behaviour';
+import type { DirectDebitFormProps } from './directDebitForm';
+import { getDirectDebitFormErrors } from './selectors';
+
+type DirectDebitFormContainerProps = {
+	render: (directDebitFormProps: DirectDebitFormProps) => JSX.Element;
+};
+
+export function DirectDebitFormContainer({
+	render,
+}: DirectDebitFormContainerProps): JSX.Element {
+	const dispatch = useContributionsDispatch();
+
+	const recaptchaCompleted = useContributionsSelector(
+		(state) => state.page.checkoutForm.recaptcha.completed,
+	);
+
+	const { countryGroupId } = useContributionsSelector(
+		(state) => state.common.internationalisation,
+	);
+
+	const {
+		accountHolderName,
+		accountNumber,
+		accountHolderConfirmation,
+		sortCodeString,
+		formError,
+	} = useContributionsSelector(
+		(state) => state.page.checkoutForm.payment.directDebit,
+	);
+
+	const errors = useContributionsSelector(getDirectDebitFormErrors);
+
+	function updateAccountHolderName(newAccountHolderName: string) {
+		dispatch(setAccountHolderName(newAccountHolderName));
+	}
+
+	function updateAccountNumber(newAccountNumber: string) {
+		dispatch(setAccountNumber(newAccountNumber));
+	}
+
+	function updateSortCode(newSortCode: string) {
+		dispatch(setSortCodeString(newSortCode));
+	}
+
+	function updateAccountHolderConfirmation(confirmed: boolean) {
+		dispatch(setAccountHolderConfirmation(confirmed));
+	}
+
+	// TODO: This functionality should be moving to a payment button in future
+	void function onSubmit() {
+		void confirmAccountDetails();
+
+		if (recaptchaCompleted) {
+			// void dispatch(payWithDirectDebit(props.onPaymentAuthorisation));
+		} else {
+			dispatch(setFormError(directDebitErrorMessages.notCompletedRecaptcha));
+		}
```

Comment:
Do we need the `if else` if we are only using the else?

Reply:
We're not using the entire function; this is left in place just as a rough overview of what needs to happen at the point of form submission.